workspaces/ui/src/components/testing/ReportSummary.js

```diff
@@ -31,67 +28,79 @@ export default function ReportSummary(props) {
   } = summary;
 
   return (
-    <div className={classes.root}>
-      <div className={classes.stats}>
-        <SummaryStats
-          totalInteractions={totalInteractions}
-          totalDiffs={totalDiffs}
-          totalUnmatchedPaths={totalUnmatchedPaths}
-        />
+    <SimulatedContext>
+      <div className={classes.root}>
+        <div className={classes.stats}>
+          <SummaryStats
+            totalInteractions={totalInteractions}
+            totalDiffs={totalDiffs}
+            totalUnmatchedPaths={totalUnmatchedPaths}
+          />
+        </div>
+
+        <small>
+          Captured from {summary.createdAt} until {summary.updatedAt}
+        </small>
+
+        <h4>Endpoints</h4>
+
+        {endpoints.length > 0 ? (
+          <ul>
+            {endpoints.map((endpoint) => (
+              <EndpointsContextStore method={endpoint.method} pathId={endpoint.pathId}>
+                <EndpointsContext.Consumer>
+                  {({endpointDescriptor, endpointId}) => {
+                    const {counts} = endpoint
+                    return (
+                      <li key={endpointId}>
+                        <PathAndMethod method={endpointDescriptor.method} path={endpointDescriptor.fullPath}/>{' '}
+                        {endpointDescriptor.endpointPurpose}: ({counts.compliant} {'/'}
+                        {counts.interactions} interactions compliant)
+                      </li>
+                    );
+                  }}
+                </EndpointsContext.Consumer>
+              </EndpointsContextStore>
+            ))}
+          </ul>
```

Comment:
I appreciate the idea of trying to re-use the endpoints domain logic to help render the endpoints correctly, but I don't think context is the right way to go here. The fact that the `EndpointContextStore` is defined and it's direct child instantly consumes is a big indicator that we're using a hammer to drive down a screw, rather than a screw driver.

Instead, I'll opt to extract the domain / view model logic from EndpointsContext as a pure function, as I've done with the view model of the summary thus far.

Using contexts this way is actually one of the major sources of confusion and barriers for me getting to grip with the UI code. In my mind, using contexts is a great place to hold application state, or to provide an interface to a service. However, domain or view-model logic, which is a lot of what I see in these contexts, deserves it's own spot,  so it can be re-used statelessly. All that _does_ belong in a Context / Store, should be accessed through a Hook. With a bit of luck, that'll prevent the endless wrapping components, super deep element trees, opaquely injected properties, etc.

The TestingDashboard feature is my exploration in how to balance these concerns in a way that  so lets keep that pattern out of it. Let's revert this to the latest what's in `feature/testing-dashboard` and I'll move on the domain logic separately, keeping this PR as an example of how the domain is used 🙂.

Reply:
I've taken this approach as part of https://github.com/opticdev/optic/pull/104/commits/307c804dac71415e91aec2415575d6a3bb9286fb, housing some of the logic that was in the now removed `APIOverviewContext` in the `ReportSummary`, ready to be factored away by using the endpoints seq in a separate PR.

Reply:
This makes sense to me, I think there have been very few occurrences when I have not directly consumed the context. I like the pattern you have here and will do something similar soon :) For now let's merge and I can have something new up here in the course of the week.

Reply:
Restored the changes in the "new style" through #105.